actor-typed-tests/src/test/java/jdocs/org/apache/pekko/typed/RouterTest.java

```diff
@@ -133,7 +133,7 @@ static Behavior<Void> showPoolRouting() {
               Routers.pool(
                   poolSize,
                   // make sure the workers are restarted if they fail
-                  Behaviors.supervise(Worker.create()).onFailure(SupervisorStrategy.restart()));
+                  Behaviors.supervise(Worker.create()).onNotFatalFailure(SupervisorStrategy.restart()));
```

Comment:
This method signature is because the overload issue between java API and scala API